### ISSUE TITLE
Compiler requires HTML template for JavaScript only module that exports a default Class

### ISSUE THREAD

**Issue by Gr8Gatsby** [salesforce/lwc#1497]
### Description
I built an LWC Open Source component that just exports a default class to help with a Tree data structure, however, the compiler wanted me to have an HTML template.

#### Steps to Reproduce

Clone this repo:
Remove HTML from: https://github.com/Gr8Gatsby/orgchart/blob/master/src/modules/my/tree/tree.html
Run `yarn watch`

Code is here:
https://github.com/Gr8Gatsby/orgchart/blob/master/src/modules/my/tree/tree.js#L25

#### Expected Results
The component is built and deployed

#### Actual Results
Compiler throws an error:

```
ERROR in ./src/modules/my/tree/tree.js
Module not found: Error: Can't resolve './tree.html' in '/Users/kevin.hill/code/orgchart/src/modules/my/tree'
 @ ./src/modules/my/tree/tree.js 1:0-32 86:8-13
 @ ./src/modules/my/org/org.js
 @ ./src/modules/my/app/app.html
 @ ./src/modules/my/app/app.js
 @ ./src/index.js
 @ multi ./node_modules/error-overlay-webpack-plugin/lib/entry-basic.js ./node_modules/error-overlay-webpack-plugin/lib/entry-devserver.js ./src/index.js
```

### Browsers Affected
All

### Version

 "dependencies": {
        "lwc-services": "^1"
    },
    "devDependencies": {
        "husky": "^2.3",
        "lint-staged": "^8.1.5"
    },
    "engines": {
        "node": ">=10.0.0"
    },

**Possible Solution**
Don't require HTML file for javaScript file that is not extending the LightningElement

**Additional context/Screenshots**

**Comment by davidturissini** [comment]
Curious, why are you including this with your modules? It seems that this code could go outside of LWC folder or you could do a named export

**Comment by Gr8Gatsby** [comment]
Fair point! Had I had my wits about me I would have just done a named import, and likely I wouldn't have hit this.

Regardless I still think that this behavior is a bug.

Seems like the only rule around having a template should be if we are extending the LightingElement.

**Comment by jodarove** [comment]
@Gr8Gatsby this have to do with https://github.com/muenzpraeger/create-lwc-app and the fact that in the project is using the `@lwc/compiler`.

The LWC compiler has 2 modes, implicit/explicit (the option `isExplicitImport`). The compiler compiles the components in "implicit" mode my default. If i remember correctly: a component is either _implicit_ or _explicit_ but never a mix of both. @diervo knows the details for sure.

The mode is controlled by lwc-create-app which is the one using the `@lwc/compiler`, therefore it should be done there. In our `@lwc/rollup-plugin` we handle such cases by providing an empty template when resolving the file.

This issue should be resolved with https://github.com/muenzpraeger/create-lwc-app/pull/104
